File: Python/inputs.py
import settings

def build_filepath(year, day, use_examples):
    root_path = settings.input_path + str(year) + settings.slash
    example_folder = settings.example_folder if use_examples == True else ''
    filename = 'Day' + two_digit_day(day) + '.txt'
    return(root_path + example_folder + filename)

def two_digit_day(day):
    #Enforce leading zero on days 1-9 for package and file names
    return '0' + str(day) if len(str(day)) == 1 else str(day)

def list_strings(filepath, separator):
    with open(filepath) as input:
        return list(map(str, input.read().split(separator)))
    
def list_ints(filepath, separator):
    with open(filepath) as input:
        return list(map(int, input.read().split(separator)))

File: CSharp/App/Inputs.cs
namespace AdventOfCode;
public static class Inputs{
    public static string BuildFilepath(int year, int day, bool useExampleFile){
        //Get the relevant filepath for the current platform from the configuration
        string platform = Settings.Platform;
        string path = Settings.Path;
        //Expand out any environment variables in the path (e.g. %UserProfile% and %HOME%)
        string expandedPath = Environment.ExpandEnvironmentVariables(path);
        string slash = (platform == "Windows") ? "\\" : "/";
        string exampleFolder = (useExampleFile == true) ? $"Examples{slash}" : "";
        string fileName = day < 10 ? $"Day0{day}.txt" : $"Day{day}.txt";

        return $"{expandedPath}Inputs{slash}{year}{slash}{exampleFolder}{fileName}";
    }
    public static List<string> ListLines(string filepath){
        return File.ReadAllLines(filepath).ToList();
    }
    public static List<string> ListStrings(string filepath, string separator){
        return File.ReadAllText(filepath).Split(separator).ToList();
    }
    public static List<int> ListInts(string filepath, string separator){
        return File.ReadAllText(filepath).Split(separator).Select(int.Parse).ToList();
    }
}